src/AzureRMTools.ts

```diff
@@ -120,72 +120,56 @@ export class AzureRMTools {
-                    if (deploymentTemplate.hasArmSchemaUri()) {
-                        foundDeploymentTemplate = true;
-
-                        this.ensureDeploymentTemplateEventsHookedUp();
-
-                        this._deploymentTemplates[documentUri] = deploymentTemplate;
-
-                        // We only initialized the stopwatch if the deployment template was being
-                        // opened. The stopwatch variable will not be initialized if the deployment
-                        // template is being edited.
-                        if (stopwatch) {
-                            // A deployment template has been opened (as opposed to having been tabbed to)
-
-                            stopwatch.stop();
-
-                            // Set the language ID to ARM deployment template
-                            if (document.languageId !== armDeploymentLanguageId) {
-                                vscode.languages.setTextDocumentLanguage(document, armDeploymentLanguageId);
-
-                                // The document will be reloaded, firing this event again with the new langid
-                                actionContext.telemetry.properties.switchedToArm = 'true';
-                                actionContext.telemetry.properties.docLangId = document.languageId;
-                                actionContext.telemetry.properties.docExtension = path.extname(document.fileName);
-                                actionContext.telemetry.suppressIfSuccessful = false;
-                                return;
-                            }
-
-                            ext.reporter.sendTelemetryEvent(
-                                "Deployment Template Opened",
-                                {
-                                    docLangId: document.languageId,
-                                    docExtension: path.extname(document.fileName),
-                                },
-                                {
-                                    documentSizeInCharacters: document.getText().length,
-                                    parseDurationInMilliseconds: stopwatch.duration.totalMilliseconds,
-                                    lineCount: deploymentTemplate.lineCount,
-                                    paramsCount: deploymentTemplate.parameterDefinitions.length,
-                                    varsCount: deploymentTemplate.variableDefinitions.length,
-                                    namespacesCount: deploymentTemplate.namespaceDefinitions.length
-                                });
-
-                            this.logFunctionCounts(deploymentTemplate);
+                    if (isNewlyOpened) {
+                        // A deployment template has been opened (as opposed to having been tabbed to)
+
+                        // Make sure the language ID is set to ARM deployment template
+                        if (document.languageId !== armDeploymentLanguageId) {
```

Comment:
What are the other possible values it can hold? I can see it having JSON. 
I'm wondering if modifying this might affect another extension that's also looking into this file.

Reply:
We've notified the extensions we're aware of that might be affected. Ultimately the user has control of this, and can turn it off our automatic behavior with a setting.  Again, good question.

This was the solution suggested by vscode guys.